src/Internal/Types/TxConstraints.purs

```diff
@@ -389,7 +445,19 @@ mustPayToScript
   -> Value
   -> TxConstraints i o
 mustPayToScript vh dt dtp vl =
-  singleton (MustPayToScript vh dt dtp Nothing vl)
+  singleton (MustPayToScript vh Nothing dt dtp Nothing vl)
+    <> guard (dtp == DatumWitness) (singleton $ MustIncludeDatum dt)
+
+mustPayToScriptAddress
```

Comment:
The naming is unfortunate imo. I seem to think we'll be tackling simplifying the number/naming of constraints in the future so doesn't matter atm

Reply:
I made it consistent with mustPayToPubKeyAddress, that is also horrible.